core/src/main/java/com/linecorp/armeria/client/SessionProtocolNegotiationException.java

```diff
@@ -34,22 +36,30 @@ public final class SessionProtocolNegotiationException extends RuntimeException
     private final SessionProtocol actual;
 
     /**
-     * Creates a new instance with the specified expected {@link SessionProtocol}.
+     * Creates a new instance with the specified expected {@link SessionProtocol} and {@link SocketAddress}.
      */
-    public SessionProtocolNegotiationException(SessionProtocol expected, @Nullable String reason) {
-        super(appendReason("expected: " + requireNonNull(expected, "expected"), reason));
+    public SessionProtocolNegotiationException(SessionProtocol expected,
+                                               @Nullable SocketAddress remoteAddress,
```

Comment:
These are breaking changes. What do you think of adding the remote address to the reason?
If `Channel` is available when raising `SessionProtocolNegotiationException`, should we set it as a hint instead? This provides users with more troubleshooting information via `Channel.toString()`.

```
"...some reason (remoteAddress: ...)"
or 
"...some reason (channel: ...)"
```